### PR TITLE
Add ESP32-S3-USB-OTG board

### PR DESCRIPTION
**Pull request by tannewt**
This board has both types of USB connectors, a display and buttons
to select items on the display. It also has a micro-B connector for
the UART output.

### CODE REVIEW

File: ports/espressif/boards/espressif_esp32s3_usb_otg_n8/pins.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    { MP_ROM_QSTR(MP_QSTR_LED_YELLOW), MP_ROM_PTR(&pin_GPIO16) },
+
+    { MP_ROM_QSTR(MP_QSTR_BUTTON_OK), MP_ROM_PTR(&pin_GPIO0) },
+    { MP_ROM_QSTR(MP_QSTR_BUTTON_UP), MP_ROM_PTR(&pin_GPIO10) },
+    { MP_ROM_QSTR(MP_QSTR_BUTTON_DW), MP_ROM_PTR(&pin_GPIO11) },
+    { MP_ROM_QSTR(MP_QSTR_BUTTON_DOWN), MP_ROM_PTR(&pin_GPIO11) },
+    { MP_ROM_QSTR(MP_QSTR_BUTTON_MENU), MP_ROM_PTR(&pin_GPIO14) },
+
+    { MP_ROM_QSTR(MP_QSTR_LCD_RST), MP_ROM_PTR(&pin_GPIO8) },
+    { MP_ROM_QSTR(MP_QSTR_LCD_EN), MP_ROM_PTR(&pin_GPIO5) },
```
Comment (gamblor21) [inline ports/espressif/boards/espressif_esp32s3_usb_otg_n8/pins.c:22]: 5 in board.c is CS. Should it be called ENable here?

Comment (tannewt) [reply ports/espressif/boards/espressif_esp32s3_usb_otg_n8/pins.c:22]: EN is the name used in the ESP docs. It functions like CS. I could add an alternate here.

Comment (gamblor21) [reply ports/espressif/boards/espressif_esp32s3_usb_otg_n8/pins.c:22]: Makes sense to me just wanted to be sure it wasn't something copy/pasted by accident.

**Review by gamblor21 — APPROVED** [review]
Looks good to me, Couldn't test

**Pull request merged**